### DISCUSSION TITLE
Question: is it possible to draw 'boxes' IN the spectrogram (using the x and y axis values)

### DISCUSSION THREAD

**Discussion by rduivenvoorde** [kitzeslab/opensoundscape#634] [Q&A] [answered]
Context: having some bat sound recordings, analyzed using software which creates x events/features for the set (as being defined as beginning at t0 ending on t1, maxfreq fmax and minfreq fmin), it would be nice if I could create spectrograms using opensoundscape to draw the 'boxes' of these features ON the spectrogram.

The goal is to compare and see the analysis of different(!) software packages: say software A finds 15 events of which a human being labeled 8 of them as certain specie calls, the other software package 'sees 20 events. The question (to us) is IF those events (seen as boxes) overlap, or not, and if we could visually use this 'diff' between the two packages to help labeling recordings.

In practice: we have some human validated bcCalls recordings which we (may be usable) as input/validation of event/feature detection of Tadarida.

Anybody has some insight IF this would already be possible? To be honest I have just slighly looked over documentation but the 'preprocessors' I see there all seem to work on a higher level (as in: combining different audio fragments), while I think my goal is to have a pointer to the actual graph, to be able to tell it: please draw a polygon between these 4 points...

Any pointer/info appreciated

**Comment by sammlapp** [comment, marked as answer]
Hi, we don't have any currently implemented features for drawing boxes on a spectrogram. However, it's pretty easy to draw boxes on a pyplot figure (over top of a spectrogram) using the frequency and time limits. Here's an example:

```
import matplotlib.patches as patches
import matplotlib.pyplot as plt
from opensoundscape.audio import Audio
from opensoundscape.spectrogram import Spectrogram

f = './audio.wav' #path to audio file

# create Spectrogram object
spec = Spectrogram.from_audio(Audio.from_file(f).trim(1,6)).bandpass(0,10000)

# Create figure and axes
fig, ax = plt.subplots()

# Display the spectrogram
mesh = ax.pcolormesh(spec.times, spec.frequencies, spec.spectrogram,shading="auto", cmap="Greys")
ax.set_xlabel("time (sec)")
ax.set_ylabel("frequency (Hz)")
cbar = fig.colorbar(mesh)
                    
#decide what the limits of the box should be
box_f = [1000, 5000]
box_t = [2.0, 3.5]

# Draw a box
rect = patches.Rectangle((box_t[0],box_f[0]),box_t[1]-box_t[0],box_f[1]-box_f[0], linewidth=1, edgecolor='b', alpha=0.3)
ax.add_patch(rect)

plt.show()
```

**Comment by rduivenvoorde** [comment]
Supercool, thanks for the code example, will try this!
